### DISCUSSION TITLE
React-native-mapbox-gl price?

### DISCUSSION THREAD

**Discussion by bmbknds** [rnmapbox/maps#1345] [Q&A] [answered]
Hello guys, 
I saw we need access token of mapbox for using library. But I don't know what feature/ service are react-native-mapbox-gl
 using from mapbox. So I dont know when will I have to pay money for mapbox, and how much. Can anyone tell me? 
 Thank you.

**Comment by ferdicus** [comment, marked as answer]
You can use this repo for free. 
However, as this repo is merely a react-native/ js wrapper around mapbox' mobile SDKs for iOS / Android, 
you'll have to consider, that, if you want to use their map tiles - you'll most probably have to pay at one point. 

Some additional information:
* With updated mobile SDKs (higher than the currently bundled versions) of `react-native-mapbox-gl` - you'll need an apiToken in order to download them - a lot of that is discussed in [this thread](https://github.com/react-native-mapbox-gl/maps/discussions/1274)
* You have the possibility to use [`MapLibre`](https://github.com/maplibre/maplibre-gl-native) instead - if you prefer an open-source flavor of mapbox. Ways of installing it are described here: [android](https://github.com/react-native-mapbox-gl/maps/blob/master/android/install.md) and here [ios](https://github.com/react-native-mapbox-gl/maps/blob/master/ios/install.md#maplibre)

**Reply by bmbknds** [reply]
Thank you for your answer. 
I'm using MapboxGL.offlineManager.createPack() for offline mode. Does it mean i'm downloading vector tiles. And I have pay money for it?
I saw in my account, I have 18/25000 user active. And it still free.

**Reply by ferdicus** [reply]
I'm no expert on Mapbox pricing - their billing structure is explained in one of the threads I liked you to. 

Their billing is two folds though:
* active users 
* tiles loaded

you can have one user that eats up all the available tiles budget or you could have too many users that don't download anything 🤷🏿
